web/main/services.py

```diff
@@ -8,3 +27,58 @@ class RemoteUser:
     full_name: str
     email: Optional[str] = None
     phone_number: Optional[str] = None
+
+
+class UserChannelService:
+
+    def __init__(self, request: 'HttpRequest', response: 'HttpResponse') -> None:
+        self.request = request
+        self.response = response
+
+    def get_ip_credentials(self) -> dict:
+        channel_settings = settings.CHANNEL_SETTINGS
+        ip_address = channel_settings.get('FORCED_IP_ADDRESS', get_remote_ip_from_request(self.request))
+        ip_server_url = channel_settings.get('GET_IP_SERVER').format(ip_address=ip_address)
+        r = requests.get(ip_server_url)
+        if not r.status_code == HTTP_200_OK:
+            logging.error(f'Ip server returned {r.status_code} {r.json()}')
+            return {'country': 'DE'}
+        return r.json()
+
+    def check_active_channels_cache_key(self) -> None:
```

Comment:
why called `check...` if you set the coockies there

Reply:
Going to rename it to set_active_channels_cache_key